p2p/src/testing_utils.rs

```diff
@@ -88,17 +79,79 @@ impl TestTransportMaker for TestTransportChannel {
 pub struct TestTransportNoise {}
 
 impl TestTransportMaker for TestTransportNoise {
-    type Transport = crate::net::mock::transport::NoiseTcpTransport;
+    type Transport = NoiseTcpTransport;
 
     type Address = SocketAddr;
 
     fn make_transport() -> Self::Transport {
-        let stream_adapter = crate::net::mock::transport::NoiseEncryptionAdapter::gen_new();
-        let base_transport = crate::net::mock::transport::TcpTransportSocket::new();
-        crate::net::mock::transport::NoiseTcpTransport::new(stream_adapter, base_transport)
+        let stream_adapter = NoiseEncryptionAdapter::gen_new();
+        let base_transport = TcpTransportSocket::new();
+        NoiseTcpTransport::new(stream_adapter, base_transport)
     }
 
     fn make_address() -> Self::Address {
         TestTransportTcp::make_address()
     }
 }
+
+/// An interface for creating peer identifiers.
+///
+/// The resulting identifier must be unique or random.
+pub trait RandomPeerIdMaker {
+    /// An identifier type.
+    type PeerId;
+
+    /// Creates a new peer identifier.
+    fn new() -> Self::PeerId;
+}
+
+pub struct TestMockPeerIdMaker {}
+
+impl RandomPeerIdMaker for TestMockPeerIdMaker {
```

Comment:
Just an nitpick.
Do we really need `RandomPeerIdMaker`? There is just one implementation of that trait and it's only used to call `MockPeerId::new()`.
And I think later backend can reuse same type for peer ids (`MockPeerId` currently).